pkg/daemon/ceph/osd/volume.go

```diff
@@ -473,14 +473,29 @@ func (a *OsdAgent) allowRawMode(context *clusterd.Context) (bool, error) {
 	return allowRawMode, nil
 }
 
-func isSafeToUseRawMode(deviceType string, cephVersion cephver.CephVersion) bool {
-	if deviceType != sys.DiskType {
-		return true
+// test if safe to use raw mode for a particular device
+func isSafeToUseRawMode(device *DeviceOsdIDEntry, cephVersion cephver.CephVersion) bool {
+	if device.DeviceInfo.Type == sys.DiskType {
+		// if this is a disk but the atari partition fix isn't in, we can't use raw mode
+		if !cephVersion.IsAtLeast(cephIgnorePhantomAtariPartitionCephVersion) {
+			logger.Debugf("won't use raw mode for disk %q since this is a disk and the atari partition issue isn't fixed", device.Config.Name)
+			return false
+		}
 	}
```

Comment:
I believe this logic is equivalent to what @satoru-takeuchi had before, just written differently to reject with `false` in these if blocks and assume true at the bottom.

From what I could tell of the previous implementation, the intent was to return true (1) if it was a partition, or (2) if it was a disk AND ceph has the atari fix.